proposals/dao/tip_118.ts

```diff
@@ -0,0 +1,198 @@
+    transferAmount.add(initialWethDepositBalance)
+  );
+
+  const resistantBalanceAndFei = await wethHoldingDeposit.resistantBalanceAndFei();
+  expect(resistantBalanceAndFei[0]).to.be.equal(transferAmount.add(initialWethDepositBalance));
+  expect(resistantBalanceAndFei[1]).to.be.equal(0);
+
+  // Withdraw ERC20
+  const receiver = '0xFc312F21E1D56D8dab5475FB5aaEFfB18B892a85';
+  const guardianSigner = await getImpersonatedSigner(addresses.pcvGuardianNew);
+  await forceEth(addresses.pcvGuardianNew);
+  await wethHoldingDeposit.connect(guardianSigner).withdrawERC20(addresses.weth, receiver, transferAmount);
+
+  expect(await wethHoldingDeposit.balance()).to.be.equal(initialWethDepositBalance);
+  expect(await contracts.weth.balanceOf(receiver)).to.be.equal(transferAmount);
+
+  // 3. Validate deprecated PSMs have no assets
+  expect(await fei.balanceOf(ethPSM.address)).to.be.equal(0);
+  expect(await weth.balanceOf(ethPSM.address)).to.be.equal(0);
+
+  expect(await fei.balanceOf(lusdPSM.address)).to.be.equal(0);
+  expect(await lusd.balanceOf(lusdPSM.address)).to.be.equal(0);
+
+  // 4. Validate transferred assets were received
+
+  // These deposits started off empty
+  // Minting currently enabled on ETH PSM, so WETH balance may increase
+  expect(await weth.balanceOf(wethHoldingDeposit.address)).to.be.at.least(EXPECTED_WETH_TRANSFER);
+  expect(await lusd.balanceOf(lusdHoldingDeposit.address)).to.be.equal(EXPECTED_LUSD_TRANSFER);
+  expect(await volt.balanceOf(voltHoldingDeposit.address)).to.be.equal(EXPECTED_VOLT_TRANSFER);
+
+  // 5. Validate deprecated PSMs have no MINTER_ROLE
+  const MINTER_ROLE = ethers.utils.id('MINTER_ROLE');
+  expect(await core.hasRole(MINTER_ROLE, addresses.ethPSM)).to.be.false;
+  expect(await core.hasRole(MINTER_ROLE, addresses.raiPriceBoundPSM)).to.be.false;
+  expect(await core.hasRole(MINTER_ROLE, addresses.lusdPSM)).to.be.false;
+
+  // 6. Validate deprecated PSMs fully paused
+  expect(await ethPSM.redeemPaused()).to.be.true;
+  expect(await ethPSM.paused()).to.be.true;
+
+  expect(await lusdPSM.redeemPaused()).to.be.true;
+  expect(await lusdPSM.paused()).to.be.true;
+
+  expect(await raiPSM.redeemPaused()).to.be.true;
+  expect(await raiPSM.mintPaused()).to.be.true;
+  expect(await raiPSM.paused()).to.be.true;
+
+  // 7. Validate PSMs are no longer safe addresses
+  expect(await pcvGuardian.isSafeAddress(ethPSM.address)).to.be.false;
+  expect(await pcvGuardian.isSafeAddress(lusdPSM.address)).to.be.false;
+  expect(await pcvGuardian.isSafeAddress(raiPSM.address)).to.be.false;
+
+  // 8. New ERC20 holding deposits are safe addresses
+  expect(await pcvGuardian.isSafeAddress(wethHoldingDeposit.address)).to.be.true;
+  expect(await pcvGuardian.isSafeAddress(lusdHoldingDeposit.address)).to.be.true;
+  expect(await pcvGuardian.isSafeAddress(voltHoldingDeposit.address)).to.be.true;
+  expect(await pcvGuardian.isSafeAddress(daiHoldingDeposit.address)).to.be.true;
```

Comment:
probably out of topic now but would be nice to have a configuration file + e2e test that lists & tests the safe addresses instead of validating here

Reply:
Yep agreed. But not needed now